### ISSUE TITLE
Patch doesn't work with MacBook (Retina, 12-inch, Early 2016)  macOS Sierra

### ISSUE THREAD

**Issue by imzhp** [Floris497/mac-pixel-clock-patch-V2#89]
Hello,Floris,thank you for the great patch.

I have a MacBook (Retina, 12-inch, Early 2016) and a Dell P2415Q   [(USB-C Digital AV Multiport Adapter)](http://www.apple.com/shop/product/MJ1K2AM/A/usb-c-digital-av-multiport-adapter?fnode=8b)
<img width="494" alt="screen shot 2016-10-15 at 15 39 29" src="https://cloud.githubusercontent.com/assets/22851660/19408365/c2ed8336-92ed-11e6-9cd4-950dc942e24f.png">

I did what you said,but failed.
<img width="714" alt="screen shot 2016-10-15 at 12 17 47" src="https://cloud.githubusercontent.com/assets/22851660/19408400/960a0154-92ee-11e6-9659-1c6f85bb3f0e.png">

<img width="446" alt="screen shot 2016-10-15 at 15 43 40" src="https://cloud.githubusercontent.com/assets/22851660/19408382/3097228e-92ee-11e6-9224-835a3505a12f.png">

I don't know the reason…
I hope you can Help；）

**Comment by Floris497** [comment]
Looks that the CoreDisplay is patched right? I will check when i get to my computer. It was working fine before i asume?

**Comment by imzhp** [comment]
Thanks,Floris.
I'm a new user and I tried so many times.
But the results are the same.

**Comment by Floris497** [comment]
So what do you want to accomplish? 4k@30Hz or 4k@60Hz?

**Comment by imzhp** [comment]
4k@60Hz

**Comment by Floris497** [comment]
You need to buy an usb-c to displayport cable. The one google sells for the chromebook pixel should work like a charm. Although I'm not sure how your macbook support the patch in 10.12 in 10.11 it works well with IOKit but in 10.12 there is CoreDisplay instead of IOKit.. and it gives black screens to some people, I've yet to investigate on that. If you don't mind spending $50 on a cable. It's your best bet to get it working :) (if it doesn't you could send back the cable within the return period I suppose)

**Comment by imzhp** [comment]
Thanks for your help,Floris.

**Comment by Floris497** [comment]
Let me know if you get something to work! With the adapter you have now you can get 4k@30Hz if not already working :)

**Comment by tiagofire** [comment]
Dear Floris,

I've tried to apply the patch to the same macbook version with MacOS Sierra. Never applied the patch before and the results were the same as described above. I just gave up on applying the patch since there are many people with so much different situations with MacOS Sierra.

My question is, based on the step the patch stopped, did it actually changed anything in the system? Do I need to restore any system files?

Thanks for your development and support here!

**Comment by Floris497** [comment]
You can disable the patch by running the command file and adding the " unpatch" to the end depending in the filw you are using something like this: `XXX-Patcher.command unpatch` there is only a small binary change in the file and the original file is backed up in the same directory.

**Comment by Woizz** [comment]
Hey guys, I just wanted to tell you, that I got the patch to work on my Macbook 12-inch (2016) using MacOS Sierra. I used DisplayCore to patch it. I get 4k@60hz on a Samsung U24E590D. I use a [Hub by Club3D, which supports HDMI 2.0](http://www.club-3d.com/index.php/products/reader.en/product/usb-30-type-c-to-hdmi-20-usb-mini-dock.html). It also allows for passthrough charging via USB-C and has a USB-A 2.0 port. If you want to get 4k@60hz over HDMI you have to have HDMI 2.0 both on your Screen and your Dongle/HUB. Of course getting a usb-c to displayport cable will probably work too, you won't be able to run in clamshell mode though and of course there's no charging while using the display.
Cheers

**Comment by Floris497** [comment]
That looks like a great adapter! happy you got it to work. I was recommending the one from google but this is far better with passthrough charging.. and it's not that much more expensive.

Enjoy!

**Comment by Woizz** [comment]
Yes, it is a great adapter. Yesterday I installed the newest Sierra Update (10.12.1) and just to make sure, everything is fine, I unpatched the CoreDisplay patch. As I rebooted my machine, I still had my Display attached and now I get 4k@60hz without any patch. I don't know if this is a hdmi 2.0 specific thing, or if apple changed that in it's newest release, but as I said, I get 4k@60hz on a 12-inch macbook (2016).
